src/moq_instance.cpp

```diff
@@ -0,0 +1,1672 @@
+                      conn_ctx.conn_id, *conn_ctx.ctrl_data_ctx_id, { .subscribe = false, .nspace = {} });
+#endif
+
+                    send_client_setup();
+                    _status = Status::READY;
+                }
+                break;
+            }
+
+            case TransportStatus::Connecting:
+                _status = Status::CLIENT_CONNECTING;
+                break;
+            case TransportStatus::RemoteRequestClose:
+                [[fallthrough]];
+
+            case TransportStatus::Disconnected: {
+
+                // Clean up publish and subscribe tracks
+                std::lock_guard<std::mutex> _(_state_mutex);
+                auto conn_it = _connections.find(conn_id);
+
+                if (conn_it == _connections.end()) {
+                    return;
+                }
+
+                for (const auto& [sub_id, delegate] : conn_it->second.tracks_by_sub_id) {
+                    delegate->cb_readNotReady(MoQTrackDelegate::TrackReadStatus::NOT_SUBSCRIBED);
+                    _delegate->cb_unsubscribe(conn_id, sub_id);
+                    remove_subscribeTrack(conn_it->second, *delegate);
+                }
+
+                for (const auto& [name_space, track] : conn_it->second.recv_sub_id) {
+                    TrackHash th(track.first, track.second);
+                    if (auto pdt = getPubTrackDelegate(conn_it->second, th)) {
+                        pdt->lock()->setSendStatus(MoQTrackDelegate::TrackSendStatus::NO_SUBSCRIBERS);
+                        pdt->lock()->cb_sendNotReady(MoQTrackDelegate::TrackSendStatus::NO_SUBSCRIBERS);
+                    }
+                }
+
+                conn_it->second.recv_sub_id.clear();
+                conn_it->second.tracks_by_sub_id.clear();
+
+                // TODO(tievens): Clean up publish tracks
+
+#ifndef LIBQUICR_WITHOUT_INFLUXDB
+                _mexport.del_conn_ctx_info(conn_id);
+#endif
+                _connections.erase(conn_it);
+
+                break;
+            }
+            case TransportStatus::Shutdown:
+                _status = Status::NOT_READY;
+                break;
+        }
+    }
+
+    void MoQInstance::on_new_connection(const TransportConnId& conn_id, const TransportRemote& remote)
+    {
+        auto [conn_ctx, is_new] = _connections.try_emplace(conn_id, ConnectionContext{});
+
+        _logger->info << "New connection conn_id: " << conn_id
+                      << " remote ip: " << remote.host_or_ip
+                      << " port: " << remote.port
+                      << std::flush;
+
+        conn_ctx->second.conn_id = conn_id;
+    }
+
+    void MoQInstance::on_recv_stream(const TransportConnId& conn_id,
+                        uint64_t stream_id,
+                        std::optional<DataContextId> data_ctx_id,
+                        const bool is_bidir)
+    {
+        auto stream_buf = _transport->getStreamBuffer(conn_id, stream_id);
+
+        //std::lock_guard<std::mutex> _(_state_mutex);
```

Comment:
Is this intentional?

Reply:
Yes, locking is applied during the operation, so it was redundant.  I had it there initially and I want to go back to see if we should move the locking around a little bit.   

I'll update the comment.   Now that it's working, I would like to see if we can further to optimize the locking in a different PR.